scipy/special/_multiufuncs.py

```diff
@@ -0,0 +1,640 @@
+                *ufunc_arg_shapes, ufunc.nout)
+
+            ufunc_arg_dtypes = tuple((ufunc_arg.dtype if hasattr(ufunc_arg, 'dtype')
+                else np.dtype(type(ufunc_arg))) for ufunc_arg in ufunc_args)
+            if hasattr(ufunc, 'resolve_dtypes'):
+                ufunc_dtypes = ufunc_arg_dtypes + ufunc.nout * (None,)
+                ufunc_dtypes = ufunc.resolve_dtypes(ufunc_dtypes) 
+                ufunc_out_dtypes = ufunc_dtypes[-ufunc.nout:]
+            else:
+                ufunc_out_dtype = np.result_type(*ufunc_arg_dtypes)
+                if (not np.issubdtype(ufunc_out_dtype, np.inexact)):
+                    ufunc_out_dtype = np.float64
+
+                ufunc_out_dtypes = ufunc.nout * (ufunc_out_dtype,)
+
+            if self.force_complex_output:
+                ufunc_out_dtypes = tuple(np.result_type(1j, ufunc_out_dtype)
+                    for ufunc_out_dtype in ufunc_out_dtypes)
+
+            out = tuple(np.empty(ufunc_out_shape, dtype = ufunc_out_dtype)
+                for ufunc_out_shape, ufunc_out_dtype
+                in zip(ufunc_out_shapes, ufunc_out_dtypes))
+
+            ufunc_kwargs['out'] = out
+
+        return ufunc(*ufunc_args, **ufunc_kwargs)
+
+
+sph_legendre_p = MultiUFunc(sph_legendre_p,
+    r"""sph_legendre_p(n, m, phi, *, diff_n=0)
+
+    Spherical Legendre polynomial of the first kind.
+
+    Parameters
+    ----------
+    n : array_like, int
+        Order of the Legendre polynomial, must have ``n >= 0``.
+    m : array_like, int
+        Degree of the Legendre polynomial.
+    phi : array_like, float
+        Input value.
+    diff_n : Optional[int]
+        A non-negative integer. Compute and return all derivatives up
+        to order ``diff_n``.
+
+    Returns
+    -------
+    p : ndarray or tuple[ndarray]
+        The spherical Legendre polynomial with ``diff_n`` derivatives.
+
+    Notes
+    -----
+    With respect to the associated Legendre polynomial :math:`P_{m}^{n}(\cos \phi)`,
+    the spherical Legendre polynomial is defined as
+
+    .. math::
+
+        \sqrt{\frac{(2 n + 1) (n - m)!}{4 \pi (n + m)!}} P_{n}^{m}(\cos \phi)
+
+    This is the same as the spherical harmonic :math:`Y_{m}^{n}(\theta, \phi)`
+    with :math:`\theta = 0`.
+    """, diff_n=0
+)
+
+
+@sph_legendre_p.override_key
+def _(diff_n):
+    diff_n = _nonneg_int_or_fail(diff_n, "diff_n", strict=False)
+    if not 0 <= diff_n <= 2:
+        raise ValueError(
+            "diff_n is currently only implemented for orders 0, 1, and 2,"
+            f" received: {diff_n}."
+        )
+    return diff_n
+
+
+sph_legendre_p_all = MultiUFunc(sph_legendre_p_all,
+    r"""sph_legendre_p_all(n, m, phi, *, diff_n=0)
+
+    Spherical Legendre polynomial of the first kind.
+
+    Parameters
+    ----------
+    n : array_like, int
+        Order (max) of the Legendre polynomial, must have ``n >= 0``.
+    m : array_like, int
+        Degree (max) of the Legendre polynomial, must have ``m >= 0``
```

Comment:
I'd mentioned this offline, but for the `all` functions, when true, we need to document that the tables are for all orders $0 \leq i \leq n$ and all degrees $-m \leq j \leq m$, and that the values are packed into the arrays in such a way that negative indexing works as expected.

Reply:
Have done so.